src/typing/typeload.ml

```diff
@@ -318,14 +318,19 @@ let rec load_params ctx info params p =
 			let c = mk_class ctx.m.curmod ([],name) p (pos e) in
 			c.cl_kind <- KExpr e;
 			TInst (c,[]),pos e
+		| TPType (CTPath({ path = { tpackage = ["$"]; tname = "_hx_default" }}),p) ->
+			(match def with
+				| Some def -> def,p
+				| None -> raise_typing_error "Cannot apply default type parameter on non-default type parameter" p
```

Comment:
I feel like this is impossible to make sense of... how about "Invalid default, type parameter %s has no default value".

Reply:
Yeah was trying to find a proper error message while reviving that branch, but had no idea.
That one should do I suppose

Reply:
I guess I need more tests now that I've added a couple error messages.. :sweat_smile: